exchanges/kraken/kraken_wrapper.go

```diff
@@ -1470,29 +1470,28 @@ func (k *Kraken) FormatExchangeKlineInterval(in kline.Interval) string {
 }
 
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (k *Kraken) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	if err := k.ValidateKline(pair, a, interval); err != nil {
-		return kline.Item{}, err
-	}
-	ret := kline.Item{
-		Exchange: k.Name,
-		Pair:     pair,
-		Asset:    a,
-		Interval: interval,
+func (k *Kraken) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := k.GetKlineRequest(pair, a, interval, start, end)
+	if err != nil {
+		return nil, err
 	}
-	candles, err := k.GetOHLC(ctx, pair, k.FormatExchangeKlineInterval(interval))
+	candles, err := k.GetOHLC(ctx,
```

Comment:
Please put in a todo and asset type check. Kraken has seperate endpoint for futures candles that is not implemented in GCT